registry/storage/driver/gcs/gcs.go

```diff
@@ -80,25 +83,42 @@ type driver struct {
 // Required parameters:
 // - bucket
 func FromParameters(parameters map[string]interface{}) (storagedriver.StorageDriver, error) {
-
 	bucket, ok := parameters["bucket"]
 	if !ok || fmt.Sprint(bucket) == "" {
 		return nil, fmt.Errorf("No bucket parameter provided")
 	}
 
-	keyfile, ok := parameters["keyfile"]
-	if !ok {
-		keyfile = ""
-	}
-
 	rootDirectory, ok := parameters["rootdirectory"]
 	if !ok {
 		rootDirectory = ""
 	}
+
+	var ts oauth2.TokenSource
+	var jwtConf *jwt.Config
+	if keyfile, ok := parameters["keyfile"]; ok {
+		jsonKey, err := ioutil.ReadFile(keyfile.(string))
```

Comment:
I think we want to check the result of the type assertion here and return an error. Otherwise, a malformed config file would panic and dump a stack trace.